src/nih_wayfinding/app/scripts/profiles/profile-model.js

```diff
@@ -8,20 +8,131 @@
     'use strict';
 
     /* ngInject */
-    function ProfileProvider () {
-        var Profile = function() {};
-        Profile.prototype.username = '';
+    function ProfileProvider (localStorageService) {
+        var Profile = function(obj) {
+            if (obj.username) {
+                this.username = obj.username;
+                this.locations = obj.locations || [];
+                this.preferences = obj.preferences || {};
+                this.tempLocation = obj.tempLocation || null;
+                this.transient = false;
+            } else {
+                this.username = '';
+                this.locations = [];
+                this.preferences = {};
+                this.tempLocation = obj.tempLocation || null;
+                this.transient = true;
+            }
+            return this;
+        };
+
+        /**
+         * Governs the persistence in localstorage of the object on which this method is called
+         */
+        Profile.prototype.save = function() {
+            if (this.transient && localStorageService.get(this.username)) {
+                return false;
+            }
+            this.transient = false;
+            localStorageService.set(this.username, this);
+            return true;
+        };
+
+        /**
+         * Sets a user preference
+         *
+         * @param property {string} key
+         * @param value {any} value
+         */
+        Profile.prototype.setPreference = function(property, value) {
```

Comment:
In general, I opted for setters out of fear that we would need to extend (rather than set) arbitrary properties. I wanted to ensure that the difference was explicit. In the actual use, I avoid directly setting except in top-level properties.